pkg/v1/tkg/client/upgrade_region.go

```diff
@@ -111,6 +111,18 @@ func (c *TkgClient) UpgradeManagementCluster(options *UpgradeClusterOptions) err
 		return err
 	}
 
+	// Validate required environment variables are set
+	log.Infof("Validating for the required environment variables to be set")
+	infraProviderName, err := getInfraNameFromRegionContext(regionalClusterClient)
+	if err != nil {
+		return errors.Wrap(err, "Unable to get infra provider from the context")
+	}
+
+	err = c.ValidateEnvVariables(infraProviderName)
+	if err != nil {
+		return errors.Wrap(err, "required env variables are not set")
+	}
+
```

Comment:
Two comments here:
1. This adds length and complexity to an already giant method.
2. This change is not covered by any tests.

Reply:
I've refactored this code. There isn't an obvious way to add a new unit test for this change within the upgrade_region code